source/frontend/src/features/documents/DocumentPreviewContainer.tsx

```diff
@@ -0,0 +1,167 @@
+import moment from 'moment';
+import { FC, useCallback, useContext, useEffect, useState } from 'react';
+
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { ModalContext } from '@/contexts/modalContext';
+import { ApiGen_Mayan_FilePage } from '@/models/api/generated/ApiGen_Mayan_FilePage';
+import { ApiGen_System_HttpStatusCode } from '@/models/api/generated/ApiGen_System_HttpStatusCode';
+
+import { DocumentViewerContext } from './context/DocumentViewerContext';
+import { IDocumentPreviewViewProps } from './DocumentPreviewView';
+import { createFileDownload } from './DownloadDocumentButton';
+import { useDocumentProvider } from './hooks/useDocumentProvider';
+
+export interface LoadedPages extends ApiGen_Mayan_FilePage {
+  loadedDocumentImageDataUrl?: string;
+  error: string;
+}
+
+export interface IDocumentPreviewContainerProps {
+  View: React.FC<IDocumentPreviewViewProps>;
+  createFileUrl?: (obj: Blob | MediaSource) => string;
```

Comment:
This looks like is always necessary, why is it marked as optional here?

Reply:
I don't mind changing this, but I struggle to see why the above approach is objectively better.

Reply:
I think is a problem of where the implementation of the injection happens. Right now we don't have a good place for that to happen. I think it would be ok if it is done on the component itself as a default vs on the function

Reply:
updated do default within the props.